File: leaoai_python_refatorado_descompactado/app/models/Factory/project.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.core.database import Base
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .projectusuario import ProjectUsuario
    from .projectstatus import ProjectStatus
    from .projectitem import ProjectItem
    from .client import Client
    from .supplier import Supplier

class Project(Base):
    __tablename__ = "projects"

    Id = Column(Integer, primary_key=True, index=True)
    Name = Column(String)
    CreatedAt = Column(DateTime)
    FinishedAt = Column(DateTime, nullable=True)
    DatePrevisioned = Column(DateTime, nullable=True)
    Observations = Column(String, nullable=True)
    PdfReportTotal = Column(String, nullable=True)
    Guid = Column(String)
    IsTemplate = Column(Boolean)
    Excluido = Column(Boolean)
    ClientId = Column(Integer, nullable=True, ForeignKey('clients.Id'))
    SupplierId = Column(Integer, nullable=True, ForeignKey('suppliers.Id'))
    ProjectStatusId = Column(Integer, nullable=True, ForeignKey('projectstatuses.Id'))

    # Define relationships here
    Client = relationship('Client', backref='projects')
    Supplier = relationship('Supplier', backref='projects')
    ProjectStatus = relationship('ProjectStatus', backref='projects')
    ProjectUsuarios = relationship('ProjectUsuario', backref='project')
    ProjectItems = relationship('ProjectItem', backref='project')

File: Framework.Data/Models/Factory/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Framework.Data.Models.Factory
{
	// PROJECT
	public class Project
	{
		[Key]
		public int Id { get; set; }

		[Required]
		public string Name { get; set; }

		[ForeignKey("ProjectStatus")]
		public int ProjectStatusId { get; set; }
		public virtual ProjectStatus ProjectStatus { get; set; }

		[ForeignKey("Client")]
		public int ClientId { get; set; }
		public virtual Client Client { get; set; }

		[ForeignKey("Supplier")]
		public int SupplierId { get; set; }
		public virtual Supplier Supplier { get; set; }

		public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
		public DateTime? FinishedAt { get; set; }
		public DateTime? DatePrevisioned { get; set; }

		public string Observations { get; set; }
		public string PdfReportTotal { get; set; }
        public string Guid { get; set; }

        public string ProductName { get; set; }
        public string Acabamento { get; set; }
        public string Codigo { get; set; }

        public bool Ativo { get; set; } = true;
		public bool Excluido { get; set; } = false;
		public bool IsTemplate { get; set; } = false;

		// Navegação
		public virtual ICollection<ProjectItem> ProjectItems { get; set; } = new List<ProjectItem>();
		public virtual ICollection<ProjectUsuario> ProjectUsuarios { get; set; } = new List<ProjectUsuario>();
	}
}
